tests/integration/test_writes/test_writes.py

```diff
@@ -1309,3 +1309,27 @@ def test_table_v1_with_null_nested_namespace(session_catalog: Catalog, arrow_tab
 
     # We expect no error here
     session_catalog.drop_table(identifier)
+
+
+@pytest.mark.integration
+def test_overwrite_all_data_with_filter(session_catalog: Catalog) -> None:
+    schema = Schema(
+        NestedField(1, "id", StringType(), required=True),
+        NestedField(2, "name", StringType(), required=False),
+        identifier_field_ids=[1],
+    )
+
+    data = pa.Table.from_pylist(
+        [
+            {"id": "1", "name": "Amsterdam"},
+            {"id": "2", "name": "San Francisco"},
+            {"id": "3", "name": "Drachten"},
+        ],
+        schema=schema.as_arrow(),
+    )
+
+    identifier = "default.test_overwrite_all_data_with_filter"
+    tbl = _create_table(session_catalog, identifier, data=[data], schema=schema)
+    tbl.overwrite(data, In("id", ["1", "2", "3"]))
+
+    assert len(tbl.scan().to_arrow()) == 3
```

Comment:
nit: since all data match the filter, the `overwrite` operation is a no-op, right? if so, can we assert that in the test? maybe show that the files are the same

Reply:
It's not a no-op, it's deleting the whole file. The change is in the delete method, not in the overwrite method.

I believe that testing the behavior is enough.

Reply:
ah, I see. The change is to make `delete` a no-op.
Sequence of operation
* pass in `overwrite_filter which matches the entire table
* in `delete`, the `overwrite_filter` is inversed, `preserve_row_filter`
* use `preserve_row_filter` on data files. 
* if the result is empty, then we don't include this data file in deletion

Reply:
Yes exactly. One thing to note is that it would be even more correct to add this to a `DELETE` snapshot, it is not replaced, but just dropped.